bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/scheduler/ExpressionThreadPoolManager.java

```diff
@@ -225,8 +225,9 @@ public void schedule(final Runnable task, final Expression expression) {
 
         @Override
         public boolean remove(Runnable task) {
-            if (futures.get(task) != null && futures.get(task).cancel(false)) {
-                running.remove(task);
+            Future<?> future = futures.get(task);
+            if (future != null && future.cancel(false)) {
+                running.remove(future);
```

Comment:
The runnable that is reached in `beforeExecute`is:
```
java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@358f17ce
```
The one that is reached into `schedule` is
```
org.eclipse.smarthome.core.scheduler.ExpressionThreadPoolExecutorTest$1@6f55137
```